synapse/tests/test_model_infotech.py

```diff
@@ -1518,6 +1501,51 @@ async def test_it_reveng(self):
     async def test_infotech_cpes(self):
 
         async with self.getTestCore() as core:
+            self.eq(r'foo:bar', core.model.type('it:sec:cpe').norm(r'cpe:2.3:a:foo\:bar:*:*:*:*:*:*:*:*:*')[1]['subs']['vendor'])
+
+            with self.raises(s_exc.BadTypeValu):
+                nodes = await core.nodes('[it:sec:cpe=asdf]')
+
+            with self.raises(s_exc.BadTypeValu):
+                await core.callStorm('[ it:sec:cpe="cpe:2.3:a:vendor001:product-foo" :v2_2="cpe:/a:vendor:product\\foo" ]')
```

Comment:
what part of this operation is raising the badTypeValu?

Reply:
The validation of the 2.2 string fails because backslash is not valid in 2.2 strings. It's failing to validate against the 2.2 regex.